src/java/io/bazel/rulesscala/worker/Worker.java

```diff
@@ -0,0 +1,114 @@
+package io.bazel.rulesscala.worker;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.lang.SecurityManager;
+import java.security.Permission;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.nio.charset.StandardCharsets;
+
+import com.google.devtools.build.lib.worker.WorkerProtocol;
+
+public final class Worker {
+
+    public static interface Interface {
+	public void work(String[] args) throws Exception;
+    }
+
+    public static void workerMain(String workerArgs[], Interface workerInterface) throws Exception {
+	if (workerArgs.length > 0 && workerArgs[0].equals("--persistent_worker")) {
+
+	    System.setSecurityManager(new SecurityManager() {
+		    @Override
+		    public void checkPermission(Permission permission) {
+			Matcher matcher = exitPattern.matcher(permission.getName());
+			if (matcher.find())
+			    throw new ExitTrapped(Integer.parseInt(matcher.group(1)));
+		    }
+		});
+
+	    InputStream stdin = System.in;
+	    PrintStream stdout = System.out;
+	    PrintStream stderr = System.err;
+	    ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+	    PrintStream out = new PrintStream(outStream);
+
+	    System.setIn(new ByteArrayInputStream(new byte[0]));
```

Comment:
Why is this needed?

Reply:
And also how does it work with you not calling `System.in` later on to get this input stream?

Reply:
I'll add a comment, but basically: there is no stdin. So we attach stdin to an empty buffer and call it a day.

If I ever get around to finishing the standalone worker tests it would make this more obvious.

Reply:
Comment added.